src/main/java/org/shanerx/tradeshop/commands/CommandRunner.java

```diff
@@ -0,0 +1,765 @@
+            sendMessage(Message.PLAYER_NOT_FOUND.getPrefixed());
+            return;
+        }
+
+        boolean removed = shop.removeManager(new ShopUser(target, ShopRole.MANAGER));
+        if (!removed) {
+            sendMessage(Message.UNSUCCESSFUL_SHOP_MEMBERS.getPrefixed());
+            return;
+        }
+
+        shop.saveShop();
+        shop.updateSign();
+        sendMessage(Message.UPDATED_SHOP_MEMBERS.getPrefixed());
+    }
+
+    /**
+     * Adds the specified player to the shop as a member
+     */
+    public void addMember() {
+        Block b = pSender.getTargetBlock(null, Setting.MAX_EDIT_DISTANCE.getInt());
+        Sign s;
+
+        if (plugin.getListManager().isInventory(b.getType())) {
+            s = findShopSign(b);
+        } else if (ShopType.isShop(b)) {
+            s = (Sign) b.getState();
+        } else {
+            sendMessage(Message.NO_SIGHTED_SHOP.getPrefixed());
+            return;
+        }
+		Shop shop = Shop.loadShop(s);
+
+		if (!shop.getOwner().getUUID().equals(pSender.getUniqueId())) {
+			sendMessage(Message.NO_SHOP_PERMISSION.getPrefixed());
+			return;
+		}
+
+        OfflinePlayer target = Bukkit.getOfflinePlayer(command.getArgAt(1));
+        if (!target.hasPlayedBefore()) {
+            sendMessage(Message.PLAYER_NOT_FOUND.getPrefixed());
+            return;
+        }
+
+        if (!shop.getManagersUUID().contains(target.getUniqueId())) {
+            shop.addMember(new ShopUser(target, ShopRole.MEMBER));
+        }
+
+        shop.saveShop();
+        shop.updateSign();
+        sendMessage(Message.UPDATED_SHOP_MEMBERS.getPrefixed());
+    }
+
+    /**
+     * Removes the specified player from the shop if they are a member
+     */
+    public void removeMember() {
+        Block b = pSender.getTargetBlock(null, Setting.MAX_EDIT_DISTANCE.getInt());
+        Sign s;
+
+        if (plugin.getListManager().isInventory(b.getType())) {
+            s = findShopSign(b);
+        } else if (ShopType.isShop(b)) {
+            s = (Sign) b.getState();
+        } else {
+            sendMessage(Message.NO_SIGHTED_SHOP.getPrefixed());
+            return;
+        }
+		Shop shop = Shop.loadShop(s);
+
+		if (!shop.getOwner().getUUID().equals(pSender.getUniqueId()) || !shop.getManagersUUID().contains(pSender.getUniqueId())) {
+			sendMessage(Message.NO_SHOP_PERMISSION.getPrefixed());
+			return;
+		}
+
+        OfflinePlayer target = Bukkit.getOfflinePlayer(command.getArgAt(1));
+        if (!target.hasPlayedBefore()) {
+            sendMessage(Message.PLAYER_NOT_FOUND.getPrefixed());
+            return;
+        }
+
+        boolean removed = shop.removeMember(new ShopUser(target, ShopRole.MEMBER));
+        if (!removed) {
+            sendMessage(Message.UNSUCCESSFUL_SHOP_MEMBERS.getPrefixed());
+            return;
+        }
+
+
+        shop.saveShop();
+        shop.updateSign();
+        sendMessage(Message.UPDATED_SHOP_MEMBERS.getPrefixed());
+    }
+
+
+}
```

Comment:
The reason i ended up with this was because I didn't want to have 12 different objects that would only be used to run a command and had no need to return anything as it would cause unnecessary overhead for the system. Also decided that while this is similar to how it was it is much easier to find specific sections and follow the code as its not all in nested if statements that just meld together.